packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1406,7 +1399,7 @@ String _generateFakePubspec(
         }
         continue;
       }
-      final String version = kManuallyPinnedDependencies[package];
+      final String version = kManuallyPinnedDependencies[package]!;
```

Comment:
I think this can be avoided with a `forEach`
```dart
    kManuallyPinnedDependencies.forEach((String package, String version) { 
      // Don't add pinned dependency if it is not in the set of all transitive dependencies.
      if (!allTransitive.contains(package)) {
        if (verbose) {
          globals.printStatus('Skipping $package because it was not transitive');
        }
        return;
      }
      result.writeln('  $package: $version');
      if (verbose) {
        globals.printStatus('  - $package: $version');
      }
    });
```

Reply:
Done